src/deadline/job_attachments/download.py

```diff
@@ -741,6 +814,42 @@ def get_output_manifests_by_asset_root(
     return outputs
 
 
+def get_output_manifest_files_by_asset_root_with_last_modified(
+    s3_settings: JobAttachmentS3Settings,
+    output_manifest_paths: list[str],
+    session: Optional[boto3.Session] = None,
+) -> dict[str, list[Tuple[datetime, BaseAssetManifest]]]:
```

Comment:
This interface would be more flexible if it returns a list that exactly mirrors the provided `output_manifest_paths`, like list[tuple[str, datetime, BaseAssetManifest]]`. If the caller doesn't actually want them grouped by asset root, they have to undo this grouping with some tricky code. On the other hand, if a caller does want the grouping, and is given the list of 3-tuples, it's a simple transformation to perform that grouping.